source/game.c

```diff
@@ -1958,72 +1982,128 @@ static void played_cards_update_loop(bool* discarded_card, int* played_selection
+
+                                    if (*played_selections > 0)
                                     {
-                                        JokerObject *joker = list_get(jokers, k);
-                                        if (joker != NULL)
+                                        for (int k = joker_scored_index; k < list_get_size(jokers); k++)
                                         {
-                                            joker->joker->processed = false; // Reset the joker's processed state for the next score
+                                            joker_scored_index += 1;
+                                            JokerObject *joker = list_get(jokers, k);
+                                            if (joker_object_score(joker, played[*played_selections - 1]->card, JOKER_CALLBACK_ON_CARD_SCORED, &chips, &mult, NULL, &money, &retrigger)) // NULLs aren't implemented yet
+                                            {
+                                                last_joker_scored_rarity = joker->joker->rarity;
+
+                                                display_chips(chips);
+                                                display_mult(mult);
+                                                display_money(money);
+
+                                                return; 
+                                            }
                                         }
                                     }
 
-                                    tte_set_pos(fx2int(played[j]->sprite_object->x) + 8, SCORED_CARD_TEXT_Y); // Offset of 16 pixels to center the text on the card
-                                    tte_set_special(0xD000); // Set text color to blue from background memory
+                                    // Score card
 
-                                    // Write the score to a character buffer variable
-                                    char score_buffer[INT_MAX_DIGITS + 2]; // for '+' and null terminator
-                                    snprintf(score_buffer, sizeof(score_buffer), "+%d", card_get_value(played[j]->card));
-                                    tte_write(score_buffer);
+                                    scored_card_index += 1; // Count the number of cards that have been scored
 
-                                    *played_selections = scored_cards;
-                                    card_object_shake(played[j], SFX_CARD_SELECT);
+                                    if (j <= played_top && card_object_is_selected(played[j]))
+                                    {
+                                        tte_set_pos(fx2int(played[j]->sprite_object->x) + 8, SCORED_CARD_TEXT_Y); // Offset of 16 pixels to center the text on the card
+                                        // TODO fix magic number
+                                        tte_set_special(TTE_BLUE_PB * 4096); // Set text color to blue from background memory
```

Comment:
```suggestion
                                        tte_set_special(TTE_BLUE_PB * TTE_SPECIAL_PB_MULT_OFFSET); // Set text color to blue from background memory
```
And add the new constant to graphic_utils.h after the TTE_..._PB defines

```
#define TTE_SPECIAL_PB_MULT_OFFSET 0x1000
```

Using the hex value is much better here because it illustrates the resulting number format. I'm not sure this is the best name for the constant, I still need to dig into the TTE color formatting to fully understand it but since you already touched this, it will get rid of the magic number.

Batch - "Add TTE_SPECIAL_PB_MULT_OFFSET"